src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1327,7 +1360,56 @@ MasterAggregateExpression(Aggref *originalAggregate, AttrNumber *columnId)
 	const AttrNumber argumentId = 1; /* our aggregates have single arguments */
 
 	if (aggregateType == AGGREGATE_COUNT && originalAggregate->aggdistinct &&
-		CountDistinctErrorRate != DISABLE_DISTINCT_APPROXIMATION)
+		CountDistinctErrorRate == DISABLE_DISTINCT_APPROXIMATION &&
+		walkerContext->repartitionSubquery)
+	{
+		Aggref *aggregate = (Aggref *) copyObject(originalAggregate);
+		List *aggTargetEntryList = aggregate->args;
+		TargetEntry *distinctTargetEntry = linitial(aggTargetEntryList);
+		List *varList = pull_var_clause_default((Node *) distinctTargetEntry->expr);
+		ListCell *varCell = NULL;
+		List *uniqueVarList = NIL;
+		int startColumnCount = walkerContext->columnId;
+
+		/* first determine number of unique var's */
+		foreach(varCell, varList)
+		{
+			Var *var = (Var *) lfirst(varCell);
+			uniqueVarList = list_append_unique(uniqueVarList, var);
+		}
+
+		/* we will be adding that many columns */
+		walkerContext->columnId += list_length(uniqueVarList);
+
+		/* reset unique var list, update each var's attribute number with new column no */
+		uniqueVarList = NIL;
+
+		foreach(varCell, varList)
+		{
+			Var *columnToUpdate = (Var *) lfirst(varCell);
+			ListCell *uniqueVarCell = NULL;
+			int columnIndex = 0;
+
+			foreach(uniqueVarCell, uniqueVarList)
+			{
+				Var *currentVar = (Var *) lfirst(uniqueVarCell);
```

Comment:
not really, we are trying to find 2 things here
1 - index of the column if it is already processed,
2 - next index value if this column is seen for the first time.

list interface has a function to check if an object is a list member, but it does not provide position information.